### PR TITLE
Mixed

### PR DESCRIPTION
**Pull request by abergeron**
- Test for #5469
- Disable the mkl warning on conda if we do find a working blas.

### CODE REVIEW

File: theano/tensor/tests/test_subtensor.py [hunk 1]
```diff
@@ -491,6 +491,8 @@ def test_ok_list(self):
                           # Test with TensorConstant index.
                           (rand(4, 2, 3),
                            theano.tensor.constant([3, 3, 1, 1, 2, 2, 0, 0])),
+                          # Test with non-contiguous input
+                          (rand(6, 2)[::2], [1, 0])
```
Comment (nouiz) [inline theano/tensor/tests/test_subtensor.py]: This do not generate non-contiguous index. [1, 0] will be converted to a numpy array that will be contibuous.

I think you need to make a new test just for that.

Comment (abergeron) [inline theano/tensor/tests/test_subtensor.py]: We don't support non-contiguous index.

Comment (nouiz) [inline theano/tensor/tests/test_subtensor.py]: The goal of the new test is to test when the graph generate a non-contiguous index, do we work correctly? Do we add the gpu_contiguous to assert we are contiguous in the graph?

This was the bug we had. I don't think this new test detect that. The first fix added an gpu_contiguous on the input, not just the index, but we looked together at the code and it wasn't requested. So you removed that from the PR. https://github.com/Theano/Theano/pull/5469/files only add gpu_contiguous for the index, not for the input.

File: theano/configdefaults.py [hunk 2]
```diff
@@ -1327,6 +1328,10 @@ def default_blas_ldflags():
         if res:
             return res
 
+        for warn in warn_record:
```
Comment (nouiz) [inline theano/configdefaults.py]: This commit seem good as it only silence the warning when we can reuse numpy blas. By to help the maintenance, can you add this comment that explain what we want to do or similar:

    If we are using conda and can't reuse numpy blas, then doing the fallback and test -lblas could give slow computation, so warn about this. But if we can link to numpy blas like with nomkl packages of conda, don't work, it should be a good enough blas.

**Comment by abergeron** [comment]
Ok, test is better now.

**Comment by slefrancois** [comment]
jenkins test this

File: theano/sandbox/cuda/tests/test_basic_ops.py [hunk 3]
```diff
@@ -1093,6 +1093,9 @@ def test_adv_sub1_fast(self):
             self.assertTrue(val.ndim == data.ndim)
             utt.assert_allclose(val, good)
 
+    def test_noncontiguous_idx(self):
+        raise SkipTest("test doesn't work here")
```
Comment (nouiz) [inline theano/sandbox/cuda/tests/test_basic_ops.py]: THe problem was that the test always use float64.

But this is the old back-end, so ok.

**Comment by nouiz** [comment]
jenkins don't test the last commit. so merging now.

**Pull request merged**